shell/platform/android/io/flutter/plugin/localization/LocalizationPlugin.java

```diff
@@ -136,4 +164,24 @@ public void sendLocalesToFlutter(@NonNull Configuration config) {
 
     localizationChannel.sendLocales(locales);
   }
+
+  private static Locale localeFromString(String localeString) {
+    // Use Locale.forLanguageTag if available (API 21+).
+    if (Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
+      return Locale.forLanguageTag(localeString);
+    } else {
+      // Pre-API 21, we fall back to manually parsing the locale tag.
+      String parts[] = localeString.split("-", -1);
+      String languageCode = parts[0];
+      String countryCode = parts.length > 1 ? parts[1] : null;
+      String scriptCode = parts.length > 2 ? parts[2] : null;
```

Comment:
I believe scriptCode comes before countrycode in locale strings. Though I'd have to double check to see the exact spec. It should be something like country/region code is 2char, while scriptcode is 4char. In most instances, the scriptcode has come first, eg, `zh-Hans-cn`

There is also divergent behavior with using `-` and `_`. I'd replace one to the other to handle both for robustness.

Reply:
You're right, I got confused by the order of the country code in the constructor.

Fixed this as suggested!